patches/tModLoader/Terraria/ModLoader/CombinedHooks.cs

```diff
@@ -36,6 +36,15 @@ public static class CombinedHooks
 			PlayerLoader.OnMissingMana(player, item, neededMana);
 		}
 
+		public static bool CanConsumeAmmo(Player player, Item weapon, Item ammo) {
+			return PlayerLoader.CanConsumeAmmo(player, weapon, ammo) & ItemLoader.CanConsumeAmmo(weapon, ammo, player);
+		}
```

Comment:
CanX hooks are to use short-circuiting. 
```suggestion
		public static bool CanConsumeAmmo(Player player, Item weapon, Item ammo) {
			return PlayerLoader.CanConsumeAmmo(player, weapon, ammo) && ItemLoader.CanConsumeAmmo(weapon, ammo, player);
		}
```

Reply:
What about CanUseItem, it uses a single `&` too

Reply:
The fact that official TML examples misused a hook is utterly bad. All CanX hooks should short-circuit on all levels, unless their return values are nullable. For DualUseWeapon, there should be something else.